### PR TITLE
tickets/DM-23880

### PR DESCRIPTION
**Pull request by mareuter**
I was able to test all of the variants locally except the one needing a persistent volume claim. After much head-banging, I gave up trying to test locally. If that feature runs into problems in actual deployment, I'll fix it at that point. Otherwise, I think the first attempt at a general CSC chart is ready to go.

### CODE REVIEW

File: csc/templates/csc-pvc.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{{- if .Values.mountpoint }}
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: {{ .Release.Name }}-pvc
+  {{- if .Values.mountpoint.gid }}
+  annotations:
+    pv.beta.kubernetes.io/gid: "{{ .Values.mountpoint.gid }}"
```
Comment (cbanek) [inline csc/templates/csc-pvc.yaml:8]: I think there might also be a uid one you can use here too, if you end up having problems.  I just like debugging it as I go, as you say

File: csc/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          stdin: true
+          tty: true
+          {{- if .Values.env }}
+          env:
+          {{- range $env_var, $env_value := .Values.env }}
+            - name: {{ $env_var }}
+              value: {{ $env_value | quote }}
+          {{- end }}
+          {{- end }}
+          {{- if or .Values.entrypoint .Values.mountpoint }}
```
Comment (cbanek) [inline csc/templates/deployment.yaml:25]: if or?  I'm not sure I get it, does it do an or on the two right things?

File: csc/values-dsm1-pvc.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+image:
+  repository: lsstts/dsm
+  tag: develop
+  pullPolicy: IfNotPresent
```
Comment (cbanek) [inline csc/values-dsm1-pvc.yaml:4]: if you're pulling develop, and that's a tag you keep pushing to, you might want to say pullPolicy: Always

File: csc/templates/csc-pvc.yaml (hunk 1 above)
Comment (mareuter) [reply csc/templates/csc-pvc.yaml:8]: Sounds like a good idea, so I'll add it.

File: csc/values-dsm1-pvc.yaml (hunk 3 above)
Comment (mareuter) [reply csc/values-dsm1-pvc.yaml:4]: I use Always up in the ArgoCD configuration.

File: csc/templates/deployment.yaml (hunk 2 above)
Comment (mareuter) [reply csc/templates/deployment.yaml:25]: or is a function in the template language so the syntax looks weird if you are used to "standard" programming languages. I know I was thrown off by this.

**Pull request merged**